app/stores/data-source.ts

```diff
@@ -0,0 +1,62 @@
+import create, { StateCreator, StoreApi } from "zustand";
+
+import { DataSource } from "@/configurator";
+import {
+  DEFAULT_DATA_SOURCE,
+  getURLParam,
+  parseDataSource,
+  stringifyDataSource,
+} from "@/domain/datasource";
+import { isRunningInBrowser } from "@/lib/is-running-in-browser";
+
+type DataSourceStore = {
+  dataSource: DataSource;
+  setDataSource: (value: string) => void;
+};
+
+const customPersist =
```

Comment:
Could you add a docstring explaining the behavior, how things are stringified in localStorage and how is it stringified in the router (if I am not mistaken, they were stringified differently, maybe it'd be the opportunity to store the same thing in localStorage and urlParam).